integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractITBatchIndexTest.java

```diff
@@ -61,7 +61,19 @@ void doIndexTestTest(
         fullDatasourceName
     );
 
+    long startSubTaskCount = -1;
+    final boolean assertRunsSubTasks = taskSpec.contains("index_parallel");
+    if (assertRunsSubTasks) {
+      startSubTaskCount = countCompleteSubTasks(fullDatasourceName);
+    }
+
+
     submitTaskAndWait(taskSpec, fullDatasourceName, waitForNewVersion);
+
+    if (assertRunsSubTasks) {
+      final long newSubTasks = countCompleteSubTasks(fullDatasourceName) - startSubTaskCount;
+      Assert.assertTrue("new sub tasks : " + newSubTasks, newSubTasks > 0);
```

Comment:
The error message should be more detailed. How about `The supervisor task[taskId] didn't create any sub tasks. Was it executed in the parallel mode?`